File: 6 kyu/Sum of Digits/solution.ex
defmodule Kata do
    def digital_root(n) when n < 10, do: n
    def digital_root(n), do: Integer.digits(n) |> Enum.sum |> digital_root
end

________________________________
defmodule Kata do
  def digital_root(n), do: rem(n - 1, 9) + 1
end

________________________________
defmodule Kata do
  def digital_root(n) do
    to_charlist(n)
      |> Enum.map(fn i -> i - 48 end)
      |> Enum.sum
      |> recurse
  end
  
  defp recurse(n) when n > 9 do
    digital_root n
  end
  defp recurse(n), do: n
end

________________________________
defmodule Kata do
    def digital_root(n) do
        String.codepoints("#{n}")
        |>sum
    end
    def sum([a]), do: String.to_integer(a)
    def sum(l)do
      Enum.map(l, &String.to_integer/1)
      |> Enum.sum
      |> inspect
      |> String.codepoints
      |> sum
    end
end

________________________________
defmodule Kata do
    def digital_root(n) do
        # Recursion limit check
        if n < 10 do
            n
        else
            # Recursive call
            curr_sum = n |> Integer.digits |> Enum.sum
            digital_root(curr_sum)
        end
    end
end

________________________________
defmodule Kata do
    def digital_root(n) when n < 10 do
      n
    end

    def digital_root(n) do
      digital_root(Integer.to_string(n)
      |> String.split("")
      |> Enum.filter(fn x -> x != "" end)
      |> Enum.map(fn x -> String.to_integer(x) end)
      |> Enum.sum)
    end
end

File: 6 kyu/Sum of Digits/solution.cpp
int digital_root(int Z) {
    return --Z % 9 + 1;
}

________________________________
int digital_sum(int n)
{
  int sum = 0;
  while (n > 0)
  {
    sum += n % 10;
    n /= 10;
  }
  
  return sum;
}

int digital_root(int n)
{
  while (n >= 10)
    n = digital_sum(n);  
  return n;
}

________________________________
int digital_root(int n)
{
  return (n-1) % 9 +1;
}

________________________________
int digital_root(int n)
{
  int digitRoot = 0;
  while(n)
  {
    digitRoot += n%10;
    n = n/10;
  }
  
  return (digitRoot > 9) ? digital_root(digitRoot) : digitRoot;
}

________________________________
int digital_root(int n)
{
  if(n < 10) 
    return n;
  return digital_root(n % 10 + digital_root(n / 10));
}

________________________________
#include <numeric>                                          //^^
int digital_root(int n){
  if (n / 10 == 0)
    return n % 10;
  std::string s = std::to_string(n);
  return digital_root(accumulate(s.begin(), s.end(), 0, [](auto a, auto b) {return a + (b-'0');}));
}
